package.json

```diff
@@ -30,6 +30,7 @@
     "handlebars": "^4.7.7",
     "marked": "^4.0.12",
     "moment": "^2.24.0",
+    "n-readlines": "^1.0.1",
```

Comment:
It was last update a year ago -- https://www.npmjs.com/package/n-readlines -- use it as it handles buffering and file handles. More efficient in reading files using less memory -- review found here https://geshan.com.np/blog/2021/10/nodejs-read-file-line-by-line/#n-readlines and other articles.

Reply:
I understand the need to use the Node.js built in - the article shows its nreadline is about 50% less memory. In the article; Default Readline 6.33Mb (stream) and nReadline 4.11Mb (chunking)

Talking with the code team, we have seen 1 client with 100 millions lines of code and a few others with 10mil lines of code every day being scanned.

Hence I would be worried about the performance and memory usage.

Reply:
Let me run some performance test and tell you.

Reply:
If we have this requirement, we'll need some long-term performance tests for it. Otherwise there's no proof that it's providing real value for us and anyone in the future can easily change something and break that use-case without knowing it.

If we really do need n-readlines, then Nebula will need to consider forking it and owning maintenance as it's currently not being maintained.

Reply:
I switched over to Node.js Built in one